### PR TITLE
skip tests which are either hanging or not supported now on rocm

### PR DESCRIPTION
**Pull request by bmedishe**

### CODE REVIEW

**Comment by bmedishe** [comment]
retest this please

File: torchrec/test_utils/__init__.py [hunk 1]
```diff
@@ -22,7 +23,7 @@
 
 TParams = ParameterSpecification("TParams")
 TReturn = TypeVar("TReturn")
-
+TEST_WITH_ROCM: bool = os.getenv("TORCHREC_TEST_WITH_ROCM", "0") == "1"
```
Comment (amathews-amd) [inline torchrec/test_utils/__init__.py:26]: The default value of this is 0. So, the CI script needs this env variable set, when testing on AMD gpu.

Comment (bmedishe) [reply torchrec/test_utils/__init__.py:26]: @amathews-amd I updated the CI script https://github.com/ROCmSoftwarePlatform/rocAutomation/blob/pytorch_torchrec_ci/pytorch/torchrec_ci/build_test_torchrec.sh#:~:text=Run%20torchrec%20unittests-,TORCHREC_TEST_WITH_ROCM%3D1,-pytest%20torchrec/datasets

Comment (amathews-amd) [reply torchrec/test_utils/__init__.py:26]: CI seems to have failed

File: torchrec/distributed/tests/test_model_parallel_hierarchical.py [hunk 2]
```diff
@@ -247,6 +247,7 @@ def test_embedding_tower_nccl(
             Dict[str, Tuple[Type[torch.optim.Optimizer], Dict[str, Any]]]
         ],
     ) -> None:
+        pdb.set_trace()
```
Comment (amathews-amd) [inline torchrec/distributed/tests/test_model_parallel_hierarchical.py]: Remove debugging

File: torchrec/distributed/tests/test_quant_model_parallel.py [hunk 3]
```diff
@@ -307,6 +308,7 @@ def test_quant_pred_state_dict(self, output_type: torch.dtype) -> None:
     )
     @settings(verbosity=Verbosity.verbose, max_examples=2, deadline=None)
     def test_quant_pred_shard(self, output_type: torch.dtype) -> None:
+        from torchrec.distributed.shard import shard_modules
```
Comment (amathews-amd) [inline torchrec/distributed/tests/test_quant_model_parallel.py:311]: is all `shard_modules` broken on rocm ?

Comment (bmedishe) [reply torchrec/distributed/tests/test_quant_model_parallel.py:311]: Error seen 
```  File "/workspace/torchrec/torchrec/distributed/tests/test_quant_model_parallel.py", line 311, in test_quant_pred_shard
    from torchrec.distributed.shard import shard_modules
  File "/workspace/torchrec/torchrec/distributed/shard.py", line 13, in <module>
    from torch.distributed._composable.contract import contract
ModuleNotFoundError: No module named 'torch.distributed._composable'
```

torch.distributed._composable is missing on rocm pytorch ( 1.13.0a0+git941769a)

**Pull request merged**